src/controller/CHIPDevice.cpp

```diff
@@ -158,6 +160,31 @@ CHIP_ERROR Device::SendCommands(app::CommandSender * commandObj)
     return commandObj->SendCommandRequest(mDeviceId, mFabricIndex, mSecureSession);
 }
 
+CHIP_ERROR Device::NewCommandSender(app::CommandSender ** commandSender)
```

Comment:
I don't see the value of having this little wrapper to allocate a `CommandSender`, which in turn, calls into a delegate to allocate one.

Callers should just allocate a CommandSender in the way they see fit (std::make_unique, new, on stack).